tests/filecheck/dialects/builtin/parse_with_location.mlir

```diff
@@ -0,0 +1,11 @@
+// RUN: xdsl-opt %s --print-op-generic | filecheck %s
+// CHECK: module
+"builtin.module"() ({
+  "func.func"() ({
+    // CHECK: ^{{.*}}(%{{.*}}: i32):
```

Comment:
Am I understanding correctly that this checks only for parsing and not printing? Do we want a test for printing also?

Reply:
Tests for printing happen in `test_printer.py`. The reason they are done separately is that right now this printing and parsing is only here for compatibility purposes. It does not add any information to the IR.